kclvm/tools/src/LSP/src/request.rs

```diff
@@ -239,3 +216,62 @@ pub(crate) fn handle_document_symbol(
     }
     Ok(res)
 }
+
+/// Called when a `Rename` request was received.
+pub(crate) fn handle_rename(
+    snapshot: LanguageServerSnapshot,
+    params: lsp_types::RenameParams,
+    sender: Sender<Task>,
+) -> anyhow::Result<Option<lsp_types::WorkspaceEdit>> {
+    // 1. check the new name validity, todo
+    // let new_name = params.new_name;
+
+    // 2. find all the references of the symbol
+    let file = file_path_from_url(&params.text_document_position.text_document.uri)?;
+    let path = from_lsp::abs_path(&params.text_document_position.text_document.uri)?;
+    let db = snapshot.get_db(&path.into())?;
+    let kcl_pos = kcl_pos(&file, params.text_document_position.position);
+    let log = |msg: String| log_message(msg, &sender);
+    let references = find_refs(
+        &db.prog,
+        &kcl_pos,
+        &db.scope,
+        snapshot.word_index_map.clone(),
+        Some(snapshot.vfs.clone()),
+        log,
+    );
+    match references {
+        core::result::Result::Ok(locations) => {
+            match locations.len() {
+                0 => {
+                    let _ = log("Symbol not found".to_string());
```

Comment:
Why use `let _` here?

Reply:
use `let _ = ...` to ignore the resulting value

the logger fn will return a result to indicate if the log msg is send successfully